### ISSUE TITLE
Ability to roll back deletions

### ISSUE THREAD

**Issue by GeiserX** [gianlucam76/k8s-cleaner#43]
I would like to begin just by thanking you for this excellent software, there are not that well-maintained software stacks and leftovers consume a substantial amount of resources. This can easily paralyze a cluster.

**Is your feature request related to a problem? Please describe.**
Imagine, that by human error (or simply by design) a deletion of a resource which is seemingly not in use, results in a deletion, followed by a crash in some stack.

**Describe the solution you'd like**
As a Kubernetes administrator, I'd like to roll back the entire set of deletions, or the ability to roll back a subset of resources.

**Describe alternatives you've considered**
Using velero before executing a cleanup job, but an integrated solution in the software would be easier and less error-prone

**Additional context**
I'm a bit concerned of deleting anything important in my production clusters :)

**Comment by gianlucam76** [comment]
Thank you @GeiserX 

This request makes sense to me. I will come up with a proposal for this.

**Comment by gianlucam76** [comment]
@GeiserX thought more about this.

I am adding a field in Cleaner Spec, which points to a valid volume where Cleaner can store Kubernetes resources.

```
	// StoreResources will store full resources in this directory.
	// +optional
	StoreResourcePath string `json:"storeResourcePath,omitempty"`
```

It is an optional field, when it is set, Cleaner will store all matching resources, before any modification is done. 

For instance, I created a Cleaner instance with name _deployment-with-zero-replicas_ which was instructed to find and delete all Deployment instances with replica set to 0.

I had one such resource in my cluster in the namespace "default" named "my-deployment"

Cleaner stored it before deleting it, then deleted it.
Resource was stored:
/tmp/k8s-pruner/deployment-with-zero-replicas/default/Deployment/my-deployment.yaml

and I was able to recreate it 

```
kubectl apply -f /tmp/k8s-pruner/deployment-with-zero-replicas/default/Deployment/my-deployment.yaml
deployment.apps/my-deployment created
```
